File: Day06/Hard/Solution.c
#include <stdio.h>
#include <string.h>

int maxFreqSum(char* s) {
    int n = strlen(s);
    int vowel = 0, conso = 0;
    int freq[256] = {0};

    for (int i = 0; i < n; i++) {
        char c = s[i];
        freq[(int)c]++;
        if (c=='a' || c=='e' || c=='i' || c=='o' || c=='u') {
            if (freq[(int)c] > vowel) vowel = freq[(int)c];
        } else {
            if (freq[(int)c] > conso) conso = freq[(int)c];
        }
    }
    return vowel + conso;
}

int main() {
    char s[1000];
    scanf("%s", s);
    printf("%d\n", maxFreqSum(s));
    return 0;
}

File: Day06/Hard/Solution.py
class Solution:
    def maxFreqSum(self, s: str) -> int:
        freq = {}
        vowel, conso = 0, 0
        for c in s:
            freq[c] = freq.get(c, 0) + 1
            if c in "aeiou":
                vowel = max(vowel, freq[c])
            else:
                conso = max(conso, freq[c])
        return vowel + conso

if __name__ == "__main__":
    s = input().strip()
    sol = Solution()
    print(sol.maxFreqSum(s))
